.github/workflows/desktop-build.yml

```diff
@@ -296,74 +305,72 @@ jobs:
       matrix:
         include:
           - arch: 32
-            triplet: x86
-            cmake: Win32
-
+            vcpkg_default_triplet: x86
+            qt_version: '5.15.2'
+            cmake_generator_platform: Win32
+            qt_arch: msvc2019
           - arch: 64
-            triplet: x64
-            cmake: x64
-            append: _64
-
-    name: Windows ${{matrix.arch}}-bit
-
+            vcpkg_default_triplet: x64
+            qt_version: '6.3.0'
+            cmake_generator_platform: x64
+            qt_arch: msvc2019_64
+            qt_modules: "qt5compat qtmultimedia qtwebsockets"
+    name: Windows (${{matrix.arch}}-bit)
     needs: configure
-
     runs-on: windows-2019
-
     env:
-      QT_VERSION: '5.15.2'
-      QT_ARCH: msvc2022${{matrix.append}}
-      CMAKE_GENERATOR: 'Visual Studio 17 2022'
-
+      CMAKE_GENERATOR: 'Visual Studio 16 2019'
     steps:
       - name: Add msbuild to PATH
         id: add-msbuild
         uses: microsoft/setup-msbuild@v1.1
 
       - name: Checkout
-        uses: actions/checkout@v2
+        uses: actions/checkout@v3
         with:
           submodules: recursive
 
-      - name: Restore Qt ${{env.QT_VERSION}} ${{matrix.arch}}-bit from cache
+      - name: Restore Qt ${{matrix.qt_version}} (${{matrix.arch}}-bit) from cache
         id: cache-qt
-        uses: actions/cache@v2
+        uses: actions/cache@v1
         with:
-          key: ${{runner.os}}-QtCache-${{env.QT_VERSION}}-${{matrix.arch}}
-          path: ${{runner.workspace}}/Qt
+          key: ${{runner.os}}-QtCache-${{matrix.qt_version}}-${{matrix.arch}}
+          path: '${{github.workspace}}/Qt'
```

Comment:
shouldn't we store this in an out of the way temporary directory like runner.temp?